hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -266,83 +415,67 @@ private function removeManagedEntityRecord($name, $type) {
   }
 
   /**
-   * Removes a list of defined activity types for a given component
+   * Removes case type if no cases are attached to it
    *
-   * @param array $activityTypes
-   *   A list of activity types names to remove
-   * @param int $componentName
-   *   (e.g : CiviCase, CiviTask .. etc)
+   * @param string $caseTypeName
    */
-  private function removeActivityTypesList($activityTypes, $componentName) {
-    civicrm_api3('OptionValue', 'get', [
-      'name' => ['IN' => $activityTypes],
-      'component_id' => $componentName,
-      'option_group_id' => "activity_type",
-      'api.OptionValue.delete' => ['id' => '$value.id'],
-    ]);
+  private function removeUnusedCaseType($caseTypeName) {
+    try {
+      $isCaseAttached = civicrm_api3('Case', 'get', [
+        'sequential' => 1,
+        'case_type_id' => $caseTypeName,
+        'options' => ['limit' => 1],
+      ]);
+
+      if (empty($isCaseAttached['values'])) {
+        civicrm_api3('CaseType', 'get', [
+          'sequential' => 1,
+          'name' => $caseTypeName,
+          'api.CaseType.delete' => ['id' => '$value.id'],
+        ]);
+      }
+    } catch (CiviCRM_API3_Exception $e) {
+      // do nothing
+    }
   }
 
   /**
-   * Ensures that any non-managed activity types for a specified list
-   * of activity types belongs to a certain component do not get deleted
-   * to prevent breaking any existing data that uses them, and instead replace the newly
-   * created managed activities with them.
+   * Removes a list of case types along with its related cases
    *
-   * @param $activityTypes
-   * @param $componentName
+   * @param array $caseTypes
+   *   A list of case type names
    */
-  private function updateManagedActivityTypes($activityTypes, $componentName) {
-    // get all current activity types that belongs to the specified component
-    $componentActivityTypes = civicrm_api3('OptionValue', 'get', [
-      'sequential' => 1,
-      'name' => ['IN' => $activityTypes],
-      'component_id' => $componentName,
-      'option_group_id' => "activity_type",
-      'options' => ['limit' => 0],
-    ]);
+  private function removeCaseTypesWithData($caseTypes) {
+    foreach ($caseTypes as $caseType) {
+      $caseTypeRow = civicrm_api3('CaseType', 'get', [
+        'sequential' => 1,
+        'name' => $caseType,
+        'limit' => 1,
+      ]);
 
-    if (!empty($componentActivityTypes['values'])) {
-      $componentActivityTypes = $componentActivityTypes['values'];
-    }
-    else {
-      $componentActivityTypes = [];
-    }
+      if (!empty($caseTypeRow['id'])) {
+        // Removing all cases related to this case type , we using
+        // DAO instead of API because API deletion does not supports
+        // batch delete and this should be much faster
```

Comment:
this comment doesn't make sense to me. The code is just using a simple delete, even though the comment talks about a batch delete. Is this correct?